bundles/AdminBundle/src/Controller/Searchadmin/SearchController.php

```diff
@@ -421,19 +421,19 @@ protected function getPermittedPaths($types = ['asset', 'document', 'object'])
      */
     protected function filterQueryParam(string $query)
     {
-        if ($query == '*') {
+        if ($query === '*') {
             $query = '';
         }
 
-        $query = str_replace('&quot;', '"', $query);
-        $query = str_replace('%', '*', $query);
-        $query = str_replace('@', '#', $query);
+        $query = str_replace(['&quot;', '%', '@'], ['"', '*', '#'], $query);
         $query = preg_replace("@([^ ])\-@", '$1 ', $query);
 
         $query = str_replace(['<', '>', '(', ')', '~'], ' ', $query);
 
+        $query = preg_replace('/(?!\B"[^"]*)([\p{L}\p{Nd}_]+)(?![^"]*"\B)/u', '$1*', $query);
```

Comment:
I think a comment on what this does would be nice, since it's not really self-explanatory ;)

Reply:
https://www.youtube.com/watch?v=SWEphAZynYo